### PR TITLE
exa: 0.9.0 -> unstable-2021-01-14

### PR DESCRIPTION
**Pull request by moni-dz**
###### Motivation for this change
exa hasn't seen a release for almost 2 years, and has had new features since then

###### Things done
update exa to latest master commit

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114550` at 1e1cfd1d run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/a7c64aca5e2a83994bd64e2f6a690457">exa</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/tools/misc/exa/default.nix:1:33:
```
  |
1 | { lib, stdenv, fetchFromGitHub, fetchpatch, rustPlatform, cmake, pandoc, perl, pkg-config, zlib
  |                                 ^
```
</li>
  </ul>
</details>

**Comment by Atemu** [comment]
Related: https://github.com/ogham/exa/issues/673

Adding an exa-unstable would be fine IMO but regular exa should stay at the upstream-recommended stable release.

**Comment by SuperSandro2000** [comment]
> Adding an exa-unstable would be fine IMO but regular exa should stay at the upstream-recommended stable release.

Just update it to unstable. Maintaining two packages is not worth it here. Or why would you want to use the old version of exa?

File: pkgs/tools/misc/exa/default.nix [hunk 1]
```diff
@@ -1,43 +1,34 @@
...
+  nativeBuildInputs = [ cmake pkg-config perl installShellFiles pandoc ];
   buildInputs = [ zlib ]
     ++ lib.optionals stdenv.isDarwin [ libiconv Security ];
 
   outputs = [ "out" "man" ];
 
   postInstall = ''
-    installManPage contrib/man/exa.1
+    ${pandoc}/bin/pandoc --standalone -f markdown -t man man/exa.1.md > man/exa.1
+    ${pandoc}/bin/pandoc --standalone -f markdown -t man man/exa_colors.5.md > man/exa_colors.5
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/exa/default.nix:26]:
```suggestion
    pandoc --standalone -f markdown -t man man/exa.1.md > man/exa.1
    pandoc --standalone -f markdown -t man man/exa_colors.5.md > man/exa_colors.5
```

**Comment by Atemu** [comment]
> why would you want to use the old version of exa?

*I* wouldn't but there must be a reason why the developers haven't released the current unstable version yet and some people value stability more than I do.  
Keep in mind that this PR will make it into nixos-stable in ~3 months which will likely not have unstable updates backported to it.

There isn't any maintenance overhead here either as the only maintenance that would have to be done to the old version would be replacing it with something very similar to the unstable one once a new version becomes available.

**Comment by SuperSandro2000** [comment]
> I wouldn't but there must be a reason why the developers haven't released the current unstable version yet and some people value stability more than I do.

I take a wild guess: Lack of time.

> Keep in mind that this PR will make it into nixos-stable in ~3 months which will likely not have unstable updates backported to it.

You can backport any updates if you think they are necessary. The main reason that many updates are not backported is that we lack the man power to do it.

There is no reason here to introduce another version of exa. Just update the current stable version to an unstable one and if we need an update on stable just backport it.

**Pull request merged**